spanner/jdbc/src/main/java/com/example/spanner/jdbc/TransactionWithRetryLoopExample.java

```diff
@@ -0,0 +1,85 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner.jdbc;
+
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.jdbc.CloudSpannerJdbcConnection;
+import com.google.cloud.spanner.jdbc.JdbcSqlExceptionFactory.JdbcAbortedException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Arrays;
+
+class TransactionWithRetryLoopExample {
+
+  static void transactionWithRetryLoop() throws SQLException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project";
+    String instanceId = "my-instance";
+    String databaseId = "my-database";
+    transactionWithRetryLoop(projectId, instanceId, databaseId);
+  }
+
+  static void transactionWithRetryLoop(String projectId, String instanceId, String databaseId)
+      throws SQLException {
+    // Create a connection that has automatic retry for aborted transactions disabled.
+    String connectionUrl =
+        String.format(
+            "jdbc:cloudspanner:/projects/%s/instances/%s/databases/%s"
+                + ";retryAbortsInternally=false",
+            projectId, instanceId, databaseId);
+    long singerId = 31;
+    long albumId = 11;
+    try (Connection connection = DriverManager.getConnection(connectionUrl)) {
+      while (true) {
+        try {
+          CloudSpannerJdbcConnection spannerConnection =
+              connection.unwrap(CloudSpannerJdbcConnection.class);
+          spannerConnection.setAutoCommit(false);
+          Mutation mutationSingers =
+              Mutation.newInsertBuilder("Singers")
+                  .set("SingerId")
+                  .to(singerId)
+                  .set("FirstName")
+                  .to("Breanna")
+                  .set("LastName")
+                  .to("Fountain")
+                  .build();
+          Mutation mutationAlbums =
+              Mutation.newInsertBuilder("Albums")
+                  .set("SingerId")
+                  .to(singerId)
+                  .set("AlbumId")
+                  .to(albumId)
+                  .set("AlbumTitle")
+                  .to("No discounts")
+                  .set("MarketingBudget")
+                  .to(1000)
+                  .build();
+          spannerConnection.bufferedWrite(Arrays.asList(mutationSingers, mutationAlbums));
+          spannerConnection.commit();
+          System.out.printf(
+              "Transaction committed at [%s]%n", spannerConnection.getCommitTimestamp().toString());
+          break;
+        } catch (JdbcAbortedException e) {
+          // Transaction aborted, retry.
+          System.out.println("Transaction aborted, starting retry");
+        }
```

Comment:
It appears the assumption is that it will eventually succeed, that might not be true in all cases.  Or is it?

Reply:
It will always eventually succeed, or fail with a different error than `Aborted`. The JDBC transactions use a Spanner `TransactionManager` under the hood, which should be used with a loop like this:

https://github.com/googleapis/java-spanner/blob/c8be7b5d43c01f6170eb02c6e8cc146cb2597ecc/google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClient.java#L252-L279